lib/dep_selector/selector.rb

```diff
@@ -67,11 +67,9 @@ def find_solution(solution_constraints, valid_packages = nil)
       packages_to_include_in_solve = trim_unreachable_packages(dep_graph, solution_constraints)
 
       begin
-        Timeout::timeout(@time_bound, Exceptions::TimeBoundExceeded) do
-          # first, try to solve the whole set of constraints
-          solve(dep_graph.clone, solution_constraints, valid_packages, packages_to_include_in_solve)
-        end
-      rescue Exceptions::NoSolutionFound
+        # first, try to solve the whole set of constraints
+        solve(dep_graph.clone, solution_constraints, valid_packages, packages_to_include_in_solve)
+      rescue Exceptions::NoSolutionFound, Exceptions::TimeBoundExceededNoSolution
         # since we're here, solving the whole system failed, so add
         # the solution_constraints one-by-one and try to solve in
         # order to find the constraint that breaks the system in order
```

Comment:
Is there benefit to attempting this when we fail due to timeout in the first place?  If the situation is just a matter of complexity (which was often the case in timeouts I looked at), returning a single package as the culprit is misleading in any case.